### ISSUE TITLE
Name resolution (and new tool, mh_lint)

### ISSUE THREAD

**Issue by florianschanda** [florianschanda/miss_hit#149]
Requires: #140 
- [ ] find a sane way to deal with toolbox/builtins; ideally by being able to look at the installed MATLAB
- [ ] new module m_sem, which is invoked after parsing
- [ ] decide on a strategy on when to invoke it, especially if no project is configured
- [ ] create new entry point (mh_lint) and documentation

Elements to resolve:
- [ ] classes (including separates), their methods and enumerations
- [ ] functions
- [ ] scope
- [ ] local variables
- [ ] globals and persistents

Types (or their lack thereof) are out of scope.
